client/dashboard/sites/overview-plan-card/index.tsx

```diff
@@ -131,10 +210,54 @@ export default function PlanCard( { site }: { site: Site } ) {
 	);
 }
 
-function getCardDescription( plan?: Plan, purchase?: Purchase ) {
-	if ( plan?.product_slug === DotcomPlans.FREE_PLAN ) {
+export default function PlanCard( { site }: { site: Site } ) {
+	const { data: plan, isLoading: isLoadingPlan } = useQuery( siteCurrentPlanQuery( site.ID ) );
+	const { data: purchase, isLoading: isLoadingPurchase } = useQuery( {
+		...sitePurchaseQuery( site.ID, plan?.id ?? '' ),
+		enabled: !! plan?.id,
+	} );
+
+	if ( isSelfHostedJetpackConnected( site ) ) {
+		return (
+			<JetpackPlanCard
+				site={ site }
+				purchase={ purchase }
+				isLoading={ isLoadingPlan || isLoadingPurchase }
+			/>
+		);
+	}
+
+	return (
+		<WpcomPlanCard
+			site={ site }
+			purchase={ purchase }
+			isLoading={ isLoadingPlan || isLoadingPurchase }
+		/>
+	);
+}
+
+function getCardDescription( site: Site, purchase?: Purchase ) {
+	if ( site.plan?.product_slug === DotcomPlans.FREE_PLAN ) {
 		return __( 'Upgrade to access all hosting features.' );
 	}
 
-	return purchase?.expiry_message;
+	if ( site.plan?.product_slug === DotcomPlans.JETPACK_FREE ) {
+		return getJetpackProductsForSite( site ).length > 0
+			? __( 'Manage subscriptions.' )
```

Comment:
To be honest, this sentence (with period / full stop) seems off. It does not look like link; it's like a broken unfinished sentence... it does not encourage user to click...

<img width="363" height="166" alt="Image" src="https://github.com/user-attachments/assets/9578af5b-c321-4a9b-80c4-e263cc0acc5d" />

Perhaps `Visit Jetpack Cloud to manage subscriptions` ?

Anyway! Not a blocker at all; feel free to skip this. Just food for thought. cc: @matt-west